frame/composable-traits/src/undercollateralized_loans/configs.rs

```diff
@@ -0,0 +1,299 @@
+	AssetId: Clone + Eq + PartialEq,
+	BlockNumber: Clone + Eq + PartialEq,
+	VaultId: Clone + Eq + PartialEq,
+{
+	/// The account id of this market
+	account_id: AccountId,
+	/// The owner of this market.
+	manager: AccountId,
+	/// The vault containing the borrow asset.
+	borrow_asset_vault: VaultId,
+	/// The asset being used as collateral.
+	collateral_asset_id: AssetId,
+	/// The asset being used as borrow asset.
+	borrow_asset_id: AssetId,
+	/// Number of blocks until invalidate oracle's price.
+	max_price_age: BlockNumber,
+	/// Borrowers which are allowed to use the service.
+	whitelist: BTreeSet<AccountId>,
+}
+
+impl<AccountId, AssetId, BlockNumber, VaultId>
+	MarketConfig<AccountId, AssetId, BlockNumber, VaultId>
+where
+	AccountId: Clone + Eq + PartialEq,
+	AssetId: Clone + Eq + PartialEq,
+	BlockNumber: Clone + Eq + PartialEq,
+	VaultId: Clone + Eq + PartialEq,
+{
+	pub fn new(
+		account_id: AccountId,
+		manager: AccountId,
+		borrow_asset_vault: VaultId,
+		borrow_asset_id: AssetId,
+		collateral_asset_id: AssetId,
+		max_price_age: BlockNumber,
+		whitelist: BTreeSet<AccountId>,
+	) -> Self {
+		Self {
+			account_id,
+			manager,
+			borrow_asset_vault,
+			borrow_asset_id,
+			collateral_asset_id,
+			max_price_age,
+			whitelist,
+		}
+	}
+
+	/// Get a reference to the market config's account id.
+	pub fn account_id(&self) -> &AccountId {
+		&self.account_id
+	}
+
+	/// Get a reference to the market config's manager.
+	pub fn manager(&self) -> &AccountId {
+		&self.manager
+	}
+	/// Get a reference to the market config's borrow asset vault.
+	pub fn borrow_asset_vault(&self) -> &VaultId {
+		&self.borrow_asset_vault
+	}
+
+	/// Get a reference to the market config's borrow asset.
+	pub fn borrow_asset_id(&self) -> &AssetId {
+		&self.borrow_asset_id
+	}
+
+	/// Get a reference to the market config's collateral asset.
+	pub fn collateral_asset_id(&self) -> &AssetId {
+		&self.collateral_asset_id
+	}
+
+	/// Get a reference to the market config's max price age.
+	pub fn max_price_age(&self) -> &BlockNumber {
+		&self.max_price_age
+	}
+
+	/// Get a reference to the market config's whitelist.
+	#[must_use]
+	pub fn whitelist(&self) -> &BTreeSet<AccountId> {
+		&self.whitelist
+	}
+}
+
+#[derive(Encode, Decode, Default, TypeInfo, RuntimeDebug, Clone, Eq, PartialEq)]
+pub struct LoanConfig<AccountId, AssetId, Balance, Timestamp>
+where
+	AccountId: Clone + Eq + PartialEq,
+	AssetId: Clone + Eq + PartialEq,
+	Balance: Clone + Eq + PartialEq,
+	Timestamp: Clone + Eq + PartialEq,
+{
+	/// Loan account id.
+	account_id: AccountId,
+	/// Market account id.
+	market_account_id: AccountId,
+	/// Borrower account id.
+	/// Should be whitelisted.
+	borrower_account_id: AccountId,
+	/// The asset being used as collateral.
+	collateral_asset_id: AssetId,
+	/// The asset being used as borrow asset.
+	borrow_asset_id: AssetId,
+	/// Amount of borrowed money.  
+	principal: Balance,
+	/// Amount of assets which should be putted as collateral.
```

Comment:
NIT: "Amount of assets which should be putted as collateral." -> "Amount of assets which should be used as collateral."

Reply:
fixed